src/serialchemy/polymorphic_serializer.py

```diff
@@ -11,7 +11,13 @@ def _get_identity(cls):
 def _get_identity_key(cls):
     identityColumn = cls.__mapper_args__['polymorphic_on']
     assert isinstance(identityColumn, Column)
-    return identityColumn.key
+    column_db_name =  identityColumn.key
+    for attribute_name, attribute in cls.__mapper__.c.items():
+        if attribute_name.startswith('_'):
+            continue
+        if attribute.key == column_db_name:
+            return attribute_name
+    return None
```

Comment:
Shouldn't this raise an error?

Reply:
Yes, I kind didn't saw a easy way to make this error happen in test since you need to create a class with a wrong `polymorphic_on` attribute. Should I do it like I just said or there is other way?